### PR TITLE
Fixed an error with '_locationPeriod' as a data criteria key

### PR DESCRIPTION
**Pull request by okeefm**
`'_locationPeriod'` didn't serialize nicely into `camelCase` thanks to the leading underscore, so special-casing transform.

Pull requests into cqm-converter require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [ ] This pull request describes why these changes were made.
- [ ] Internal ticket for this PR:
- [ ] Internal ticket links to this PR
- [ ] Code diff has been done and been reviewed
- [ ] Tests are included and test edge cases
- [ ] Tests have been run locally and pass
- [ ] If applicable, the library version number in `cqm-converter.gemspec` has been updated

**Bonnie Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Cypress Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/cqm/converter/hds_record.rb [hunk 1]
```diff
@@ -43,7 +43,14 @@ def to_qdm(record)
       cql_qdm_patient.keys.each do |dc_type|
         cql_qdm_patient[dc_type].each do |dc|
           # Convert snake_case to camelCase
-          dc_fixed_keys = dc.deep_transform_keys { |key| key.to_s.camelize(:lower) }
+          dc_fixed_keys = dc.deep_transform_keys do |key|
```
Comment (holmesie) [inline lib/cqm/converter/hds_record.rb]: What about: `dc_fixed_keys = dc.deep_transform_keys { |key| key.to_s.gsub(/^_/, '').camelize(:lower) }`?

**Comment by dczulada** [comment]
does _code also need to be accounted for?  The facilityLocation "Code" is still uppercase.

`      "facilityLocations": [
        {
          "Code": {
            "code": "4525004",
            "codeSystem": "SNOMED-CT",
            "version": null,
            "descriptor": "Emergency Department Visit"
          },
          "locationPeriod": {
            "low": "2012-05-11T07:00:00+00:00",
            "high": "2012-05-11T08:15:00+00:00",
            "lowClosed": true,
            "highClosed": true
          }
        }`

File: lib/cqm/converter/hds_record.rb [hunk 2]
```diff
@@ -125,5 +125,17 @@ def to_qdm(record)
...
     end
+
+    def generate_qdm_data_element(dc_fixed_keys, dc_type)
+      data_element = QDM.const_get(dc_type).new.from_json(dc_fixed_keys.to_json)
+
+      if data_element.is_a?(QDM::EncounterPerformed)
+        data_element.facilityLocations = data_element.facilityLocations.map do |facility|
+          QDM::FacilityLocation.new.from_json(facility.to_json)
+        end
+      end
```
Comment (holmesie) [inline lib/cqm/converter/hds_record.rb]: I think there might be a few "nested" QDM things we might eventually need to handle here... For now, do you think adding a little comment (something along the lines of: "Any nested QDM types that need initialization should be handled here when converting from the QDM models to the HDS models") makes sense?

File: lib/cqm/converter/qdm_patient.rb [hunk 3]
```diff
@@ -156,6 +160,15 @@ def extractor(qdm_thing)
       end
     end
 
+    def location_extractor(fac)
+      inter = interval_extractor(fac[:locationPeriod].as_json.symbolize_keys)
```
Comment (holmesie) [inline lib/cqm/converter/qdm_patient.rb]: Is `interval` a reserved word? I definitely always prefer spelling these variable names out.

Comment (holmesie) [reply lib/cqm/converter/qdm_patient.rb]: Same with `fac`.

File: lib/cqm/converter/hds_record.rb [hunk 4]
```diff
@@ -125,5 +125,17 @@ def to_qdm(record)
 
       patient
     end
+
+    def generate_qdm_data_element(dc_fixed_keys, dc_type)
+      data_element = QDM.const_get(dc_type).new.from_json(dc_fixed_keys.to_json)
+
+      if data_element.is_a?(QDM::EncounterPerformed)
```
[the remark this answers is not in the document]
Comment (okeefm) [reply lib/cqm/converter/hds_record.rb]: It's the only one where the `facilityLocation` isn't a `Code` object...thanks, model info file!

File: lib/cqm/converter/hds_record.rb (hunk 2 above)
Comment (okeefm) [reply lib/cqm/converter/hds_record.rb]: Sure thing.

File: lib/cqm/converter/qdm_patient.rb (hunk 3 above)
Comment (okeefm) [reply lib/cqm/converter/qdm_patient.rb]: Yeah, I'll fix that.
